--- ogre-1.12.13/OgreMain/include/OgrePlatform.h.orig	2022-02-07 17:47:25.723366601 +0100
+++ ogre-1.12.13/OgreMain/include/OgrePlatform.h	2022-02-07 17:48:50.842905469 +0100
@@ -75,7 +75,7 @@ namespace Ogre {
 #endif
 
 /* Find the arch type */
-#if defined(__x86_64__) || defined(_M_X64) || defined(_M_X64) || defined(_M_AMD64) \
+#if ((defined(__x86_64__) || defined(_M_X64) || defined(_M_X64) || defined(_M_AMD64)) && !defined(__ILP32__)) \
  || defined(__ppc64__) \
  || defined(__arm64__) || defined(__aarch64__) || defined(_M_ARM64) \
  || defined(__mips64) || defined(__mips64_) \
--- ogre-1.12.13/OgreMain/src/OgrePlatformInformation.cpp.orig	2021-08-04 20:33:45.000000000 +0200
+++ ogre-1.12.13/OgreMain/src/OgrePlatformInformation.cpp	2022-02-07 17:47:25.726699917 +0100
@@ -118,7 +118,7 @@ namespace Ogre {
         }
     #endif
 #elif (OGRE_COMPILER == OGRE_COMPILER_GNUC || OGRE_COMPILER == OGRE_COMPILER_CLANG) && OGRE_PLATFORM != OGRE_PLATFORM_EMSCRIPTEN
-        #if OGRE_ARCH_TYPE == OGRE_ARCHITECTURE_64
+        #if OGRE_ARCH_TYPE == OGRE_ARCHITECTURE_64 || defined(__ILP32__)
            return true;
        #else
         unsigned oldFlags, newFlags;
@@ -158,7 +158,7 @@ namespace Ogre {
         result._edx = CPUInfo[3];
         return result._eax;
 #elif (OGRE_COMPILER == OGRE_COMPILER_GNUC || OGRE_COMPILER == OGRE_COMPILER_CLANG) && OGRE_PLATFORM != OGRE_PLATFORM_EMSCRIPTEN
-        #if OGRE_ARCH_TYPE == OGRE_ARCHITECTURE_64
+        #if OGRE_ARCH_TYPE == OGRE_ARCHITECTURE_64 || defined(__ILP32__)
         __asm__
         (
             "cpuid": "=a" (result._eax), "=b" (result._ebx), "=c" (result._ecx), "=d" (result._edx) : "a" (query)
@@ -232,7 +232,7 @@ namespace Ogre {
         }
     #endif
 #elif (OGRE_COMPILER == OGRE_COMPILER_GNUC || OGRE_COMPILER == OGRE_COMPILER_CLANG) && OGRE_PLATFORM != OGRE_PLATFORM_EMSCRIPTEN
-        #if OGRE_ARCH_TYPE == OGRE_ARCHITECTURE_64 
+        #if OGRE_ARCH_TYPE == OGRE_ARCHITECTURE_64 || defined(__ILP32__)
             return true;
         #else
         // Does gcc have __try/__except similar mechanism?
