diff -ur ogre_src_v1-8-1/OgreMain/include/OgrePlatform.h ogre_src_v1-8-1-x32/OgreMain/include/OgrePlatform.h
--- ogre_src_v1-8-1/OgreMain/include/OgrePlatform.h	2012-09-02 07:27:15.000000000 +0200
+++ ogre_src_v1-8-1-x32/OgreMain/include/OgrePlatform.h	2015-06-15 23:34:03.582303164 +0200
@@ -130,7 +130,7 @@
 #endif
 
     /* Find the arch type */
-#if defined(__x86_64__) || defined(_M_X64) || defined(__powerpc64__) || defined(__alpha__) || defined(__ia64__) || defined(__s390__) || defined(__s390x__)
+#if (defined(__x86_64__) && !defined(__ILP32__)) || defined(_M_X64) || defined(__powerpc64__) || defined(__alpha__) || defined(__ia64__) || defined(__s390__) || defined(__s390x__)
 #   define OGRE_ARCH_TYPE OGRE_ARCHITECTURE_64
 #else
 #   define OGRE_ARCH_TYPE OGRE_ARCHITECTURE_32
diff -ur ogre_src_v1-8-1/OgreMain/src/OgrePlatformInformation.cpp ogre_src_v1-8-1-x32/OgreMain/src/OgrePlatformInformation.cpp
--- ogre_src_v1-8-1/OgreMain/src/OgrePlatformInformation.cpp	2012-09-02 07:27:15.000000000 +0200
+++ ogre_src_v1-8-1-x32/OgreMain/src/OgrePlatformInformation.cpp	2015-06-15 23:36:47.257495085 +0200
@@ -116,7 +116,7 @@
         }
 	#endif
 #elif (OGRE_COMPILER == OGRE_COMPILER_GNUC || OGRE_COMPILER == OGRE_COMPILER_CLANG) && OGRE_PLATFORM != OGRE_PLATFORM_NACL
-        #if OGRE_ARCH_TYPE == OGRE_ARCHITECTURE_64
+        #if OGRE_ARCH_TYPE == OGRE_ARCHITECTURE_64 || defined(__ILP32__)
            return true;
        #else
         unsigned oldFlags, newFlags;
@@ -170,7 +170,7 @@
         }
 	#endif
 #elif (OGRE_COMPILER == OGRE_COMPILER_GNUC || OGRE_COMPILER == OGRE_COMPILER_CLANG) && OGRE_PLATFORM != OGRE_PLATFORM_NACL
-        #if OGRE_ARCH_TYPE == OGRE_ARCHITECTURE_64
+        #if OGRE_ARCH_TYPE == OGRE_ARCHITECTURE_64 || defined(__ILP32__)
         __asm__
         (
             "cpuid": "=a" (result._eax), "=b" (result._ebx), "=c" (result._ecx), "=d" (result._edx) : "a" (query)
